src/Widgets/AppButton.vala

```diff
@@ -140,10 +140,27 @@ public class Slingshot.Widgets.AppButton : Gtk.Button {
                 app_launched ();
             });
 
-            if (menu != null && menu.get_children () != null) {
+            if (menu.get_children () != null) {
                 menu.popup (null, null, null, e.button, e.time);
                 return true;
             }
+
+            return false;
+        });
+
+        this.key_press_event.connect ((e) => {
+            if (e.keyval == Gdk.Key.Menu) {
+                menu = new Slingshot.AppContextMenu (desktop_id, desktop_path);
```

Comment:
We probably want to DRY this whole thing and create a method like "popup_context_menu" or something yeah?

Reply:
I can do that but on mouse click the popup opens at the click position, and on menu button click it opens at calculated position on the AppButton. What do you think?

Reply:
It would probably still be worth it to make a method that optionally takes a Gtk.Widget and then have an if/else there

Reply:
Ok. Will do that.

Reply:
Extracted menu creation in a method, also replace deprecated _popup_ method with _popup_at_pointer_ .